sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -111,7 +113,10 @@ def test_lasso_zero():
     # Check that the lasso can handle zero data without crashing
     X = [[0], [0], [0]]
     y = [0, 0, 0]
-    clf = Lasso(alpha=0.1).fit(X, y)
+    # _cd_fast.pyx tests for gap < tol, but here we get 0.0 < 0.0
+    # should probably be changed to gap <= tol ?
+    with ignore_warnings(category=ConvergenceWarning):
```

Comment:
I am wondering if we could decorate the function with a `pytest.mark.filterwarning` instead?

Reply:
Yes, in principle we could decorate the functions that way to simplify - probably a style questions - decorations versus explicit code.

But first I want to see what is going on with `gap <= tol` in `_cd_fast.pyx`, see above.